File: NET/JNetCall.Sharp.InProc/Client/InProcClient.cs
using JNetBase.Sharp.Threads;
using JNetCall.Sharp.Client.Tools;

namespace JNetCall.Sharp.Client
{
    public static class InProcClient
    {
        public static T Create<T>(string jar) where T : class
        {
            var pool = new ThreadExecutor();
            var protocol = new JvmTransport(jar);
            var handler = new ClassProxy(protocol, pool);
            handler.Listen();
            return ClientHelper.Create<T>(handler);
        }
    }
}

File: JVM/jnetcall-java-inproc/src/main/java/jnetcall/java/client/InProcClient.java
package jnetcall.java.client;

import jnetbase.java.threads.ThreadExecutor;
import jnetcall.java.client.tools.ClientHelper;

public final class InProcClient {

    public static <T> T create(Class<T> clazz, String dll) {
        ThreadExecutor pool = new ThreadExecutor();
        ClrTransport protocol = new ClrTransport(dll, 15);
        ClassProxy handler = new ClassProxy(protocol, pool);
        handler.listen();
        return ClientHelper.create(clazz, handler);
    }
}
